### PR TITLE
[3.x] Proxy now properly selects proxy settings from system properties

### LINKED ISSUE #6446: Unable to create a Proxy of ProxyType.SYSTEM
## Environment Details
* Helidon Version: 2.5.0
* Helidon MP
* JDK version: 17.0.3
* OS: MacOS
* Docker version (if applicable): N/A

----------

## Problem Description
Unable to create a Proxy of ProxyType.SYSTEM. This is how I was able to workaround but it is not ideal since I'm not using the SystemSelector.
```
    static Optional<Proxy> createProxyFromSystemProperties() {
        String httpsProxy = System.getProperty("https.proxyHost");
        String host = httpsProxy == null ? System.getProperty("http.proxyHost") : httpsProxy;
        String noProxy = System.getProperty("http.nonProxyHosts");
        String proxyPort = System.getProperty("http.proxyPort");
        if (host != null) {
            Proxy.Builder proxyBuilder = Proxy.builder()
                    .type(Proxy.ProxyType.HTTP)
                    .host(host);

            if (noProxy != null) {
                Arrays.stream(noProxy.split(Pattern.quote("|")))
                        // Helidon NoProxy only works if the * is not provided
                        .map(s -> s.startsWith("*.") ? s.substring(1) : s)
                        .forEach(proxyBuilder::addNoProxy);
            }

            if (proxyPort != null) {
                proxyBuilder.port(Integer.parseInt(proxyPort));
            }
            return Optional.ofNullable(proxyBuilder.build());
        }
        return Optional.empty();
    }
```

## Steps to reproduce
I've created the following test classe in project generated by the helidon-quickstart-mp archetype to highlight the issue:
```
package io.helidon.examples.quickstart.mp;

import io.helidon.webclient.Proxy;

import java.lang.reflect.Field;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
public class SystemProxyTest {
    @Test
    void staticMethodCreationTest() throws NoSuchFieldException, IllegalAccessException {
        // The javadoc says:
        // Create from environment and system properties.
        // Returns: a proxy instance configured based on this system settings
        Proxy proxy = Proxy.create();
        Field field = proxy.getClass().getDeclaredField("type");
        field.setAccessible(true);

        assertEquals(Proxy.ProxyType.SYSTEM, field.get(proxy));
    }

    @Test
    void builderCreationTest() throws NoSuchFieldException, IllegalAccessException {
        // The javadoc says:
        // Configure proxy from environment variables and system properties.
        // Params:
        // useIt – use system selector
        Proxy proxy = Proxy.builder()
                .useSystemSelector(true)
                .build();
        Field field = proxy.getClass().getDeclaredField("type");
        field.setAccessible(true);

        assertEquals(Proxy.ProxyType.SYSTEM, field.get(proxy));
    }
}
```
I've added the official javadoc in that code snippet to highlight that the behavior is different than expected.

The culprit seems to be this condition (which still exists in Helidon 3.0):
https://github.com/helidon-io/helidon/blob/84123e8caa7f6671efc396526404b2b39d0afbbf/webclient/webclient/src/main/java/io/helidon/webclient/Proxy.java#L429

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #6446

### CODE REVIEW

File: config/test-infrastructure/src/main/java/io/helidon/config/test/infra/ResetSystemProperties.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Annotation used to reset system properties to the state before test execution.
+ */
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (tomas-langer) [inline config/test-infrastructure/src/main/java/io/helidon/config/test/infra/ResetSystemProperties.java:29]: This is not the right location for this class. Please backport module `helidon-common-testing-junit5`, and mark existing classes that are now moved as deprecated

File: webclient/webclient/pom.xml [hunk 2]
```diff
@@ -117,6 +117,12 @@
             <scope>provided</scope>
             <optional>true</optional>
         </dependency>
+        <dependency>
```
Comment (tomas-langer) [inline webclient/webclient/pom.xml:120]: This should not be here, as we are not testing config

Comment (tomas-langer) [reply webclient/webclient/pom.xml:120]: (see comment above - we need a new module)

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: config/test-infrastructure/src/main/java/io/helidon/config/test/infra/ResetSystemProperties.java (hunk 1 above)
Comment (Verdent) [reply config/test-infrastructure/src/main/java/io/helidon/config/test/infra/ResetSystemProperties.java]: backported in the new commit

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**